go/state/externalstate/external_state.go

```diff
@@ -0,0 +1,631 @@
+// Copyright (c) 2025 Sonic Operations Ltd
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at soniclabs.com/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package externalstate
+
+//go:generate sh ../../lib/build_libcarmen.sh
+
+/*
+#cgo CFLAGS: -I${SRCDIR}/../../../cpp
+#cgo LDFLAGS: -L${SRCDIR}/../../lib -lcarmen -L${SRCDIR}/../../../rust/target/release -lcarmen_rust
+#cgo LDFLAGS: -Wl,-rpath,${SRCDIR}/../../lib -Wl,-rpath,${SRCDIR}/../../../rust/target/release
+#include <stdlib.h>
+#include "state/c_state.h"
+*/
+import "C"
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"unsafe"
+
+	"github.com/0xsoniclabs/carmen/go/common/witness"
+
+	"github.com/0xsoniclabs/carmen/go/common/amount"
+	"github.com/0xsoniclabs/carmen/go/state"
+
+	"github.com/0xsoniclabs/carmen/go/backend"
+	"github.com/0xsoniclabs/carmen/go/common"
+)
+
+const CodeCacheSize = 8_000 // ~ 200 MiB of memory for go-side code cache
+const CodeMaxSize = 25000   // Contract limit is 24577
+
+type ExternalImpl int
+
+const (
+	ExternalImplCpp ExternalImpl = iota
+	ExternalImplRust
+)
```

Comment:
I guess these could and should all be not exported.

```suggestion
const codeCacheSize = 8_000 // ~ 200 MiB of memory for go-side code cache
const codeMaxSize = 25000   // Contract limit is 24577

type externalImpl int

const (
	externalImplCpp externalImpl = iota
	externalImplRust
)
```